text/0000-set-helper.md

```diff
@@ -0,0 +1,106 @@
+- Start Date: 2020-02-20
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# `set` helper
+
+## Summary
+
+This RFC introduces a new `set` helper like the one provided by
+[`ember-simple-set-helper`](https://github.com/pzuraq/ember-simple-set-helper).
+
+The `set` helper will always return a function that will change a value.
+
+```hbs
+<SelectCountry @update={{set this.country}} />
+```
+
+The new `set` helper can be used instead of the old `{{fn (mut ..)}}` combination.
+However this RFC does not deprecate `{{mut}}`.
+
+## Motivation
+
+The existing `mut` helper is quite helpful, but behaves often in an unexpected way.
+
+When used to pass an argument as in `<MyComponent @foo={{mut this.bar}}>` or
+`{{my-component foo=(mut this.bar)}}` it will do nothing for a glimmer component
+but provide a box with a `value` and an `update` function on the private `attrs`
+property of a classic component.
+
+However the more common use-case is to pass the result of `mut` to either the 
+`{{action}}` or the `{{fn}}` helper:
+
+```hbs
+<SelectCountry @update={{fn (mut this.country)}} />
+<button {{on "click" (fn (mut this.name) "")}}>delete name</button>
+```
+
+However many people would expect `{{fn (mut this.foo)}}` to be the same as
+`{{mut this.foo}}` in all cases, which can lead to confusion.
```

Comment:
Could you expand this with more detail on what exactly the difference is? This is a distinction I've loaded up into my head a number of times, but since I make a point *not* to reach for `{{mut}}` I inevitably forget it—and it's not clear to me what the actual difference is *in practice and effect* between these two. If you could show in these examples how they differ in behavior, that would help not only me but everyone else in evaluating this RFC!

Reply:
I tried to explain the problem a bit. I would love some feedback if its understandable.